apps/aens/src/aens.erl

```diff
@@ -0,0 +1,78 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2018, Aeternity Anstalt
+%%% @doc
+%%%    Resolve registered names
+%%% @end
+%%%-------------------------------------------------------------------
+
+-module(aens).
+
+%% API
+-export([resolve/2,
+         get_name_entry/1]).
+
+%%%===================================================================
+%%% Types
+%%%===================================================================
+
+-define(LABEL_SEPARATOR, <<".">>).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec resolve(atom(), binary()) -> {ok, binary()} | {error, atom()}.
+resolve(Type, Binary) ->
+    case lists:reverse(binary:split(Binary, ?LABEL_SEPARATOR, [global, trim])) of
+        [Binary] ->
+            aec_base58c:safe_decode(Type, Binary);
+        [RegistrarNamespace|_Namespace] ->
+            %% When we have more registrars, pick faster and still gov friendly structure
+            %% If we allow to purchase registrars, this will live in regular tree
+            case [RN || RN <- aec_governance:name_registrars(), RN =:= RegistrarNamespace] of
+                [] ->
+                    {error, registrar_unknown};
+                _ ->
+                    {ok, StateTrees} = aec_conductor:top_state_trees(),
```

Comment:
May I suggest not to define `top_state_trees/0` in `aec_conductor` but rather defining `aec_conductor:get_name/1` (similarly to `aec_conductor:get_account/1`)?